test/test_carma_clock.cpp

```diff
@@ -57,17 +57,58 @@ TEST(test_carma_clock, test_sim_time_initialization)
 {
     // try a sim clock
     CarmaClock clock(true);
+    // Measure start time
+    auto start = system_clock::now();
+    // Thread sleep for SYSTEM_SLEEP_TIME milliseconds before initializing clock
     std::thread t([&clock]() {
         std::this_thread::sleep_for(std::chrono::milliseconds(SYSTEM_SLEEP_TIME));
         clock.update(1);
+
     });
-    auto start = system_clock::now();
+    // blocks until clock initialized
     clock.wait_for_initialization();
+    // Measure end time
     auto after = system_clock::now();
     auto msCount = duration_cast<milliseconds>(after - start).count();
     // should have happened after the update occurred in other thread
     EXPECT_NEAR(SYSTEM_SLEEP_TIME, msCount, 10);
     t.join();
+
+}
+
+TEST(test_carma_clock, test_sim_time_initialization_multiple_threads)
+{
+    // try a sim clock
+    CarmaClock clock(true);
```

Comment:
```
CarmaClock clock(true);
clock.update(1);
```
After calling the update (1), the _current_time is 1, and   _is_initialized = true. The _initialization_cv will notify_all().
Then thread t sleep for 150 milliseconds and call update(2), update(3), update(4).
I am assuming the _current_time - 4 by now? Also wait_for_initialization() is not executed as it is only executed when the _is_initialized  = false.
So my confusion here is that how come the EXPECT_EQ(simTimeNow, 1); pass? also why do we call the 
wait_for_initialization() if it is already initialized?

Reply:
It is a bit hard to guess without being able to run the unit test myself.

Reply:
I am able to run the unit test and see the _current_time run differently from real time. Maybe we have a function to return the _current_time, and ASSERT the _current_time change instead of thesystem clock time?

Reply:
I am not sure you understand this unit test. The first thread is sleeping for .15 seconds and then calling update, the other 3 are called immediately. They are blocked for .15 seconds until the update is called from the other thread, which notifies all of them. If you add cout statements you can confirm this in the update method by checking when threads are notified. 

In other words all the code inside the 4 threads is executed in parallel.